main.go

```diff
@@ -118,3 +121,31 @@ func main() {
 		setupLog.Error(err, "Failed to wait server")
 	}
 }
+
+func kubernetesEnvironmentNeeded(cfg *config.Config) bool {
+	if cfg.Server4 != nil {
```

Comment:
You can simplify this by using the `k8s.io/apimachinery/pkg/util/sets` package:

```
// Superset as a string slice (array)
supersetArray := []string{"apple", "banana", "cherry", "date"}

// Subset as a string slice (array)
subsetArray := []string{"orange", "banana", "grape"}

// Convert superset array to a sets.String
superset := sets.NewString(supersetArray...)

// Convert subset array to a sets.String
subset := sets.NewString(subsetArray...)

// Check if there's an intersection (at least one common element)
if superset.HasAny(subset.List()...) {
	fmt.Println("Superset contains at least one element from the subset")
} else {
	fmt.Println("Superset does not contain any element from the subset")
}
```

Reply:
<3